modules/tftp/server.rb

```diff
@@ -63,6 +63,13 @@ def delete_file(file)
         logger.debug "TFTP: Skipping a request to delete a file which doesn't exists"
       end
     end
+
+    def self.outdated_files(age)
+      delete_before = Time.now - age
+      Dir.glob(File.join(Proxy::TFTP::Plugin.settings.tftproot, "boot", "*-{vmlinuz,initrd.img}")).filter do |file|
+        File.file?(file) && File.mtime(file) < delete_before
```

Comment:
Could `ctime` be used? Assuming the file is written once and then never changed, would ctime ever change?

Reply:
No, but then you still have the problem where:

* Host X provisions RHEL 9
* Host X's token expires
* Host Y enters build mode for RHEL 9
* Pruning happens
* Host Y attempts to retrieve TFTP files

Note that's also a problem with the current foreman_maintain implementation. Perhaps few users hit this because they may use external Smart Proxies and never run the code on them.

Reply:
I assume that `Host Y enters build mode for RHEL 9` does a check if the files already exist on the smart-proxy, and if they do no operation is performed. And in theory, we would need to "touch" the files somehow to indicate. Which I think then is where you were getting at Foreman (or maybe the smart-proxy object?) having a canonical list of needed files?

Reply:
When a host enters build mode (which can also be rebuilding an existing host), it immediately retrieves the files. Then the host still needs to boot. A BIOS (or UEFI) can take a significant amount of time on server class hardware (a minute is often on the low end). So if the pruning happens between those 2 events (Smart Proxy fetches TFTP files and host fetches TFTP files from Smart Proxy) you have a problem.

I think this requires a bit more thought on what the best way is, but bringing it back to the original issue: I'd recommend removing this from a health check and instead make it a separate step.